packages/server/src/service-worker-manager.ts

```diff
@@ -104,13 +112,20 @@ export class ServiceWorkerManager implements IServiceWorkerManager {
       registration = (await serviceWorker.getRegistration(scriptURL)) || null;
       // eslint-disable-next-line no-console
       console.info('JupyterLite ServiceWorker was already registered');
+    } else {
+      await this._unregisterOldServiceWorkers();
```

Comment:
We need to unregister old service workers if `serviceWorker.controller` is not defined. This means old service workers are outdated due to hard refresh.

Reply:
OK thanks for the comment.

I guess we'll be able to test this more with the next pre-releases.